lib/web/userpreferences.go

```diff
@@ -182,9 +182,16 @@ func userPreferencesResponse(resp *userpreferencesv1.UserPreferences) *UserPrefe
 }
 
 func clusterPreferencesResponse(resp *userpreferencesv1.ClusterUserPreferences) ClusterUserPreferencesResponse {
-	return ClusterUserPreferencesResponse{
-		PinnedResources: resp.PinnedResources.ResourceIds,
+	jsonResp := ClusterUserPreferencesResponse{
```

Comment:
I would do the following:

- rename the argument from `resp` to `prefs`
- now that `resp` is free, rename `jsonResp` to `resp`